src/fields/Fields.cpp

```diff
@@ -250,42 +251,78 @@ Fields::Copy (int lev, int i_slice, FieldCopyType copy_type, int slice_comp, int
     }
 
     amrex::Box const& vbx = fab.box();
-    if (vbx.smallEnd(Direction::z) <= i_slice and
-        vbx.bigEnd  (Direction::z) >= i_slice)
+    if (vbx.smallEnd(Direction::z) <= i_slice / diag_coarsen[2] and
+        vbx.bigEnd  (Direction::z) >= i_slice / diag_coarsen[2] and
+        i_slice % diag_coarsen[2] == 0)
     {
         amrex::Box copy_box = vbx;
-        copy_box.setSmall(Direction::z, i_slice);
-        copy_box.setBig  (Direction::z, i_slice);
+        copy_box.setSmall(Direction::z, i_slice / diag_coarsen[2]);
+        copy_box.setBig  (Direction::z, i_slice / diag_coarsen[2]);
 
         amrex::Array4<amrex::Real> const& full_array = fab.array();
 
         const amrex::IntVect ncells_global = geom.Domain().length();
         const bool nx_even = ncells_global[0] % 2 == 0;
         const bool ny_even = ncells_global[1] % 2 == 0;
 
-        if (copy_type == FieldCopyType::FtoS) {
+        const int coarse_x = diag_coarsen[0];
+        const int coarse_y = diag_coarsen[1];
+        const int coarse_z = diag_coarsen[2];
+
+        const int ncells_x = ncells_global[0];
+        const int ncells_y = ncells_global[1];
+
+        if (slice_dir ==-1 /* 3D data */){
+
             amrex::ParallelFor(copy_box, ncomp,
             [=] AMREX_GPU_DEVICE (int i, int j, int k, int n) noexcept
             {
-                slice_array(i,j,k,n+slice_comp) = full_array(i,j,k,n+full_comp);
+                amrex::Real field_value = 0;
+                int n_values = 0;
+                for (int j_c = 0; j_c < coarse_y && (j*coarse_y+j_c) < ncells_y; ++j_c) {
+                    for (int i_c = 0; i_c < coarse_x && (i*coarse_x+i_c) < ncells_x; ++i_c) {
+                        field_value += slice_array( i*coarse_x+i_c,
+                                                    j*coarse_y+j_c,
+                                                    k*coarse_z,
+                                                    n+slice_comp);
+                        ++n_values;
+                    }
+                }
+                full_array(i,j,k,n+full_comp) = field_value / std::max(n_values,1);
```

Comment:
If I understand correctly, this is large-order smoothing interpolation (where the stencil length is coarse_* for direction *), where the same coefficient is applies to all point of the stencil. This causes significant smoothing of the data, which could be unwanted. Besides, we probably don't have to re-write interpolation functions, we already have some (originally written for particles, but also used for field interpolation in https://github.com/Hi-PACE/hipace/pull/572).

In this case, I would recommend to do sub-sampling rather than large-stencil interpolation, where we would ideally just take 1 point in each direction. Of course, this is impractical because we want to conserve the centering of data. For this I would recommend to do as shown on the image below: interpolate between the two nearest grid points (in each direction), which is the minimum to fix the centering (this is what we do for slice I/O). Do you think you could implement this instead? Of course this is valid for x and y, and for 3D slice I/O.

![IMG_20210809_104848](https://user-images.githubusercontent.com/26292713/128681361-aaac641c-75f1-42e2-bc02-caa1f6400545.jpg)